pkg/controller/cronjob/utils.go

```diff
@@ -196,13 +196,17 @@ func (o byJobStartTime) Len() int      { return len(o) }
 func (o byJobStartTime) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
 
 func (o byJobStartTime) Less(i, j int) bool {
-	if o[j].Status.StartTime == nil {
-		return o[i].Status.StartTime != nil
-	}
-
 	if o[i].Status.StartTime.Equal(o[j].Status.StartTime) {
 		return o[i].Name < o[j].Name
 	}
 
+	if o[i].Status.StartTime == nil {
```

Comment:
I'm pretty sure you need to have this before the previous invocation.

Reply:
Are you referring to the .Equal call? It handles the nil case appropriately: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L80

Reply:
IMO it's a clearer flow to have the `if nil` up front.

Reply:
@tomwans you're right, but I still lean towards the readability what @vllry mentioned. Please move the if check before.

Reply:
:+1: